src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -161,8 +159,7 @@ public static partial class JsonSerializer
                     }
                 }
 
-                JsonEncodedText escapedKey = JsonEncodedText.Encode(key, options.Encoder);
```

Comment:
We should consider isolating this change and porting to 3.1 (based on measurements - if it shows noticeable perf improvements).